src/Sdk/WebApi/WebApi/Results/ResultsHttpClient.cs

```diff
@@ -0,0 +1,130 @@
+                step_backend_id = stepId
+            };
+
+            var stepSummaryUploadRequest = new Uri(m_resultsServiceUrl, "twirp/results.services.receiver.Receiver/GetStepSummarySignedBlobURL");
+
+            HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, stepSummaryUploadRequest);
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", m_token);
+            requestMessage.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json"));
+
+            HttpContent content = new ObjectContent<GetSignedStepSummaryURLRequest>(request, m_formatter);
+            requestMessage.Content = content;
+
+            var response = await SendAsync(requestMessage, HttpCompletionOption.ResponseContentRead, cancellationToken: cancellationToken);
+            return await ReadJsonContentAsync<GetSignedStepSummaryURLResponse>(response, cancellationToken);
+        }
+
+        public async Task StepSummaryUploadCompleteAsync(string jobId, string planId, string stepId, long size, CancellationToken cancellationToken)
+        {
+            var timestamp = DateTime.UtcNow.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK");
+            var request = new StepSummaryMetadataCreate()
+            {
+                workflow_job_run_backend_id = jobId,
+                workflow_run_backend_id = planId,
+                step_backend_id = stepId,
+                size = size,
+                uploaded_at = timestamp
+            };
+
+            var stepSummaryUploadCompleteRequest = new Uri(m_resultsServiceUrl, "twirp/results.services.receiver.Receiver/CreateStepSummaryMetadata");
+
+            HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, stepSummaryUploadCompleteRequest);
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", m_token);
+            requestMessage.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json"));
+
+            HttpContent content = new ObjectContent<StepSummaryMetadataCreate>(request, m_formatter);
+            requestMessage.Content = content;
+
+            var response = await SendAsync(requestMessage, HttpCompletionOption.ResponseContentRead, cancellationToken: cancellationToken);
+            var jsonResponse = await ReadJsonContentAsync<CreateStepSummaryMetadataResponse>(response, cancellationToken);
+            if (!jsonResponse.ok)
+            {
+                throw new Exception($"Failed to mark step summary upload as complete, status code: {response.StatusCode}, ok: {jsonResponse.ok}, jobId: {jobId}, planId: {planId}, stepId: {stepId}, size: {size}, timestamp: {timestamp}");
+            }
+        }
+
+        public static async Task<HttpResponseMessage> UploadFileAsync(string url, string blobStorageType, FileStream file, CancellationToken cancellationToken) {
+            // Upload the file to the url
+            var httpClient = new HttpClient();
```

Comment:
why do we create a new HttpClient if we are using the RawHttpClientBase? should we use the `RawHttpClientBase.SendAsync`?

Reply:
I believe the intention is this upload goes to Blob, not Results - that said, I think the `Raw` http client should still work.